validator/coordinator/src/test/java/tech/pegasys/teku/validator/coordinator/DepositProviderTest.java

```diff
@@ -41,23 +41,27 @@
 import tech.pegasys.teku.datastructures.util.OptimizedMerkleTree;
 import tech.pegasys.teku.pow.event.DepositsFromBlockEvent;
 import tech.pegasys.teku.ssz.SSZTypes.SSZList;
+import tech.pegasys.teku.ssz.SSZTypes.SSZMutableList;
 import tech.pegasys.teku.storage.client.RecentChainData;
 import tech.pegasys.teku.util.config.Constants;
 
 public class DepositProviderTest {
 
   private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
   private final RecentChainData recentChainData = mock(RecentChainData.class);
-  private final BeaconState beaconState = mock(BeaconState.class);
+  private final BeaconState state = mock(BeaconState.class);
   private final Eth1DataCache eth1DataCache = mock(Eth1DataCache.class);
-  private final MerkleTree depositMerkleTree =
-      new OptimizedMerkleTree(Constants.DEPOSIT_CONTRACT_TREE_DEPTH);
   private List<tech.pegasys.teku.pow.event.Deposit> allSeenDepositsList;
   private final DepositProvider depositProvider =
       new DepositProvider(recentChainData, eth1DataCache);
+  private final Eth1Data randomEth1Data = dataStructureUtil.randomEth1Data();
+
+  private MerkleTree depositMerkleTree;
 
   @BeforeEach
   void setUp() {
+    depositMerkleTree = new OptimizedMerkleTree(Constants.DEPOSIT_CONTRACT_TREE_DEPTH);
```

Comment:
nit: shouldn't need to move this assignment to setUp (field can be non-final with an assignment if needed).

Reply:
But in that case, we'd be adding duplicate deposit data roots to the Merkle tree between tests since multiple tests call `mockDepositsFromEth1Block`

Reply:
No a new instance of the test class is created for every test so field values are not shared between test methods.